src/lib/pickles_types/plonk_types.ml

```diff
@@ -208,6 +387,93 @@ module Features = struct
     end
   end]
 
+  let of_full
+      ({ range_check0
+       ; range_check1
+       ; foreign_field_add
+       ; foreign_field_mul
+       ; xor
+       ; rot
+       ; lookup
+       ; runtime_tables
+       ; uses_lookups = _
+       ; table_width_at_least_1 = _
+       ; table_width_at_least_2 = _
+       ; table_width_3 = _
+       ; lookups_per_row_3 = _
+       ; lookups_per_row_4 = _
+       ; lookup_pattern_xor = _
+       ; lookup_pattern_range_check = _
+       } :
+        'bool Full.t ) =
+    { range_check0
+    ; range_check1
+    ; foreign_field_add
+    ; foreign_field_mul
+    ; xor
+    ; rot
+    ; lookup
+    ; runtime_tables
+    }
+
+  let to_full ~or_:( ||| )
+      { range_check0
+      ; range_check1
+      ; foreign_field_add
+      ; foreign_field_mul
+      ; xor
+      ; rot
+      ; lookup
+      ; runtime_tables
+      } : _ Full.t =
+    let lookup_pattern_range_check =
+      (* RangeCheck, Rot gates use RangeCheck lookup pattern *)
+      range_check0 ||| range_check1 ||| rot
+    in
+    let lookup_pattern_xor =
+      (* Xor lookup pattern *)
+      xor
+    in
+    (* Make sure these stay up-to-date with the layouts!! *)
```

Comment:
Nit: I would add a comment pointing to the layout definition, i.e. in `kimchi/src/circuits/lookup/lookups.rs`, function `lookups`. And do the same in the repository proof-systems. It helps the user to know where the user has to modify the code in the different places.